File: lab_02/code.py
import time
x = 1
def step():
    x**2 - x**2 + x**4-x**5+x+x
    

if __name__ == '__main__':
    while True:
        try:
            n = int(input("Int: "))
            print(f"n={n}; solving...")
            begin = time.process_time()
            for _ in range(int(n)):
                step()
            end = time.process_time()
            elapsed = end - begin
            print(f'time per step (us): {elapsed/n * 1e6}')
        except ValueError:
            print("exit")
            break

File: lab_02/code.cpp
#include <iostream>
#include <cmath>
#include <chrono>

using namespace std;

double x = 1;

void step()
{
    pow(x,2)-pow(x,2)+pow(x,4)-pow(x,5)+x+x;
}

int main()
{
    int n;
    cout << "Int: ";
    while (true) {
        if(cin >> n){
            cout << "n=" << n << "; solving..." << endl;
            auto begin = std::chrono::high_resolution_clock::now();
            for (int i = 0; i < n; i++)
            {
                step();
            }
            auto end = std::chrono::high_resolution_clock::now();
            auto elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(end - begin);

            cout << "time per step (us): " << elapsed.count()* 1e-3 / n << endl;
            cout << "Int: ";
        } else {
            cout << "exit";
            break;
        }
    } 
    return 0;
}